controller/sort_delete_test.go

```diff
@@ -28,6 +28,22 @@ func TestFilterObjectsForDeletion(t *testing.T) {
 	}
 }
 
+func TestFilterObjectsForDeletionForKind(t *testing.T) {
+	tests := []struct {
+		input []*unstructured.Unstructured
+		want  []*unstructured.Unstructured
+	}{
+		{[]*unstructured.Unstructured{NewPod(), NewService()}, []*unstructured.Unstructured{NewService()}},
+		{[]*unstructured.Unstructured{NewPod()}, []*unstructured.Unstructured{NewPod()}},
+		{[]*unstructured.Unstructured{}, []*unstructured.Unstructured{}},
```

Comment:
Wondering if we could have a test with objects which would almost cause a "hung" situation without the deletion sequence enforcement ?

Reply:
Not sure if there is a way to reproduce a hung delete - ideas welcome!
This PR helps solve the "churn" with resources like endpoints reported during deletion.